### PR TITLE
DOC: update `pandas/core/ops.py` docstring template to accept examples

### PR DESCRIPTION
**Pull request by alysivji**
Checklist for the pandas documentation sprint

- [x] PR title is "DOC: update the <your-function-or-method> docstring"
- [x] The PEP8 style check passes: `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] It has been proofread on language by another sprint participant

The Chicago assigned docstrings uses a template with string formatting. To add examples for the various DataFrame operators, I modified the template to allow examples for each operator.

Opening this PR to get comments on this workflow.

I will work with all the ChiPy contributors to make sure we can get their examples in once we figure out the best way forward.

### CODE REVIEW

**Comment by TomAugspurger** [comment]
This one is tricky, since we have so many potential examples. I think ideally we would have a docstring stub that creates the dataframes, and then examples stubs for each of the arithmetic methods.

**Comment by TomAugspurger** [comment]
Something like (untested)


```python
_arith_frames = """\
>>> a = pd.DataFrame([1, 1, 1, np.nan], index=['a', 'b', 'c', 'd'],
                     columns=['one'])
>>> a
   one
a  1.0
b  1.0
c  1.0
d  NaN
>>> b = pd.DataFrame(dict(one=[1, np.nan, 1, np.nan],
                          two=[np.nan, 2, np.nan, 2]),
                     index=['a', 'b', 'd', 'e'])
>>> b
   one  two
a  1.0  NaN
b  NaN  2.0
d  1.0  NaN
e  NaN  2.0
"""

_add_doc_frame = """\
Examples
--------
{create}
>>> a.add(b, fill_value=0)
   one  two
a  2.0  NaN
b  1.0  2.0
c  1.0  NaN
d  1.0  NaN
e  NaN  2.0 
""".format(create=_arith_frames)
```

**Comment by jorisvandenbossche** [comment]
@alysivji Do you want us to merge this one? So it is easier for the other PRs to add the examples?

File: pandas/core/ops.py [hunk 1]
```diff
@@ -343,50 +343,93 @@ def _get_op_name(op, special):
 # -----------------------------------------------------------------------------
 # Docstring Generation and Templates
 
+_add_doc_FRAME = """
```
Comment (jorisvandenbossche) [inline pandas/core/ops.py]: maybe use "example" in the name?

**Comment by alysivji** [comment]
@TomAugspurger each team did different examples.

@jorisvandenbossche if we can come up with a best practice i think that would be best. i can help get all the examples in once we figure out the best workflow.

**Comment by TomAugspurger** [comment]
@alysivji I think you may know best here, since you've seen some of the other examples I think.

I suspect that the PRs would have many conflicts (removing the old example, updating the kwargs, etc.) and that the only distinct bits are the actual new example and updating the `df_examples` value. If that's the case, let's merge this and then start taking PRs that have an updated copy of master + the new example.

**Comment by jorisvandenbossche** [comment]
Yes, I think it is fine to merge this.

File: pandas/core/ops.py [hunk 2]
```diff
@@ -343,50 +343,93 @@ def _get_op_name(op, special):
...
+_add_example_FRAME = """
+>>> a = pd.DataFrame([1, 1, 1, np.nan], index=['a', 'b', 'c', 'd'],
+                     columns=['one'])
+>>> a
+   one
+a  1.0
+b  1.0
+c  1.0
+d  NaN
+>>> b = pd.DataFrame(dict(one=[1, np.nan, 1, np.nan],
```
Comment (jreback) [inline pandas/core/ops.py]: blank line between ases

File: pandas/core/ops.py [hunk 3]
```diff
@@ -343,50 +343,93 @@ def _get_op_name(op, special):
...
+                          two=[np.nan, 2, np.nan, 2]),
+                     index=['a', 'b', 'd', 'e'])
+>>> b
+   one  two
+a  1.0  NaN
+b  NaN  2.0
+d  1.0  NaN
+e  NaN  2.0
+>>> a.add(b, fill_value=0)
```
Comment (jreback) [inline pandas/core/ops.py]: could add some text here to indicate what the filling is doing

**Review by jreback — APPROVED** [review]
ok to merge, my comments can be addressed in the next iteration.

**Comment by jreback** [comment]
thanks @alysivji

File: pandas/core/ops.py (hunk 2 above)
Comment (kurtiskerstein) [reply pandas/core/ops.py]: @jreback I am about to put in a PR for another one of the examples.  Where should there be a blank line? Between what would be "cell" executions?

Comment (kurtiskerstein) [reply pandas/core/ops.py]: @jreback I am about to put in a PR for one of the examples.  Where should there be a blank line? Between what would be "cell" executions?

Comment (jorisvandenbossche) [reply pandas/core/ops.py]: @kurtiskerstein I think there should be only blank lines around "blocks" of code that you want to have appear together in one code block

**Pull request merged**